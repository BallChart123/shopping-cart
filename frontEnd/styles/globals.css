html,
body {
  padding: 0;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

.btn-custom {
  margin-top: 32px;
}
/* .card-img > img {
  height: auto;
}
.card-img {
  max-height: 300px;
} */
.items-product {
  margin-bottom: 20px;
}
.product-image {
  width: 100%;
}

.card-img {
  position: relative;
}
.card-img:after {
  content: "";
  display: block;
  padding-bottom: 100%; /* The padding depends on the width, not on the height, so with a padding-bottom of 100% you will get a square */
}

.card-img img {
  position: absolute; /* Take your picture out of the flow */
  top: 0;
  bottom: 0;
  left: 0;
  right: 0; /* Make the picture taking the size of it's parent */
  width: 100%; /* This if for the object-fit */
  height: 100%; /* This if for the object-fit */
  object-fit: cover; /* Equivalent of the background-size: cover; of a background-image */
  object-position: center;
}
