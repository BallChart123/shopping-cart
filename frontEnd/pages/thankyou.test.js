import PaymentSuccess from './thankyou'
import { Button } from 'react-bootstrap'
import renderer from 'react-test-renderer'
import * as nextRouter from 'next/router'
import { expect } from '@jest/globals'

// setup mock

nextRouter.useRouter = jest.fn()
nextRouter.useRouter.mockImplementation(() => ({
  route: '/thankyou',
  query: {
    orderId: '123',
    trackingId: '456',
  }
}))

nextRouter.default.push = jest.fn()

// test case

describe('PaymentSuccess', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <PaymentSuccess/>,
    )
    const json = component.toJSON()
    expect(json).toMatchSnapshot()
  })

  it('should route to index page after click OK button', () => {
    const component = renderer.create(
      <PaymentSuccess/>,
    )
    const testInstance = component.root
    testInstance.findByType(Button).props.onClick()

    expect(nextRouter.default.push).toHaveBeenCalledWith('/')
  })
})
