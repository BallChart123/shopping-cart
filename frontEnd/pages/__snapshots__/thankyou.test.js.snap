// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentSuccess should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
    >
      <h1
        id="thankyou"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Thankyou
      </h1>
    </div>
  </div>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <img
        className=""
        src="https://www.pngitem.com/pimgs/m/69-692608_transparent-answer-icon-png-check-pass-icon-png.png"
        width="8%"
      />
    </div>
  </div>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
    >
      <p
        id="success"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Successfully
      </p>
    </div>
  </div>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
    >
      <p
        id="order_id"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Order ID : 123
      </p>
    </div>
  </div>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
    >
      <p
        id="tracking_id"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Tracking ID : 456
      </p>
    </div>
  </div>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col"
    >
      <p
        id="check_email"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Please check your email
      </p>
    </div>
  </div>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-sm-4"
    >
      <button
        className="btn btn-primary btn-block"
        disabled={false}
        id="ok_button"
        onClick={[Function]}
        type="button"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
